### PR TITLE
feat: Cover grpc crud rbac

### PR DESCRIPTION
**Pull request by pasha-codefresh**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by jannfis** [comment]
Could you please add some meaningful description to this PR? What's it for, and why we'd need it? Is there an open issue?

**Comment by pasha-codefresh** [comment]
Sure @jannfis , actually it is part of https://github.com/argoproj/argo-cd/issues/2251, and it is implemented in wrong way, I already review it with Alex today. Im going to change it

**Comment by alexmt** [comment]
FYI: PR addresses this comment: https://github.com/argoproj/argo-cd/pull/6943#pullrequestreview-730920088

File: server/repository/repository.go [hunk 1]
```diff
@@ -94,14 +94,15 @@ func (s *Server) List(ctx context.Context, q *repositorypkg.RepoQuery) (*appsv1.
...
+	repo, err := s.db.GetRepository(ctx, q.Repo)
+	if err != nil {
 		return nil, err
 	}
 
-	repo, err := s.db.GetRepository(ctx, q.Repo)
-	if err != nil {
+	if err := s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceRepositories, rbacpolicy.ActionGet, createRBACObject(repo.Project, repo.Repo)); err != nil {
 		return nil, err
 	}
```
Comment (ishitasequeira) [inline server/repository/repository.go:104]: I guess returning the error of GetRepository before doing the EnforceErr check could possibly result in [unwanted information disclosure](https://cwe.mitre.org/data/definitions/200.html). 

Can we return a permission denied here maybe?

File: server/repository/repository.go [hunk 2]
```diff
@@ -349,10 +352,21 @@ func (s *Server) UpdateRepository(ctx context.Context, q *repositorypkg.RepoUpda
...
+
+	repo, err := s.db.GetRepository(ctx, q.Repo.Repo)
+	if err != nil {
+		return nil, err
+	}
+
+	// verify that user can do update inside project where repository is located
+	if err := s.enf.EnforceErr(ctx.Value("claims"), rbacpolicy.ResourceRepositories, rbacpolicy.ActionUpdate, createRBACObject(repo.Project, repo.Repo)); err != nil {
+		return nil, err
+	}
```
Comment (ishitasequeira) [inline server/repository/repository.go:364]: Same for this scenario and other scenarios below too.

File: server/repository/repository.go (hunk 1 above)
Comment (pasha-codefresh) [reply server/repository/repository.go:104]: Good comment, thank you

**Review by alexmt — APPROVED** [review]
LGTM. Thank you!

File: server/repository/repository.go [hunk 3]
```diff
@@ -51,6 +51,21 @@ func NewServer(
 	}
 }
 
+func (s *Server) getRepo(ctx context.Context, url string) (*appsv1.Repository, error) {
+	repo, err := s.db.GetRepository(ctx, url)
+	if err != nil {
+		return nil, status.Error(codes.PermissionDenied, "permission denied")
```
Comment (jannfis) [inline server/repository/repository.go:57]: Is this actually the same error that's returned by `EnforceErr`?

**Pull request merged**